### PR TITLE
chore: save payment info in vss

### PR DESCRIPTION
**Pull request by EthanYuan**
- save payment info in vss
- use `last_update` as version

### CODE REVIEW

File: mutiny-wasm/src/indexed_db.rs [hunk 1]
```diff
@@ -488,6 +489,36 @@ impl IndexedDbStorage {
...
+                                }
+                            } else {
+                                log_debug!(
+                                    logger,
+                                    "Skipping vss key {} with version {}, current version is {}",
+                                    kv.key,
+                                    kv.version,
+                                    info.last_update
+                                );
+                                return Ok(None);
```
Comment (jjyr) [inline mutiny-wasm/src/indexed_db.rs:512]: If `info.last_update` isn't outdated, shouldn't we just return info?

Comment (EthanYuan) [reply mutiny-wasm/src/indexed_db.rs:512]:
```rust
                for kv in keys {
                    futs.push(Self::handle_vss_key(kv, vss, &map, logger));
                }
                let results = futures::future::try_join_all(futs).await?;

                let mut items_vector = Vec::with_capacity(results.len());
                for (key, value) in results.into_iter().flatten() {
                    // save to memory and batch the write to local storage
                    map.write_data(key.clone(), value.clone(), None)?;
                    items_vector.push((key, value));
                }
```

Based on the context `fn read_all`, the current logic is designed to handle only keys that need updates. Returning `Ok(None)` when `info.last_update` is not outdated aligns with this intent, as the upper-level logic uses `results.into_iter().flatten()` to discard these None values and ensure only updated data is processed.

**Review by jjyr — APPROVED** [review]

**Comment by duanyytop** [comment]
Should this PR also implement the function of get_activity to obtain data from VSS?

**Comment by EthanYuan** [comment]
> Should this PR also implement the function `get_activity` of `mutiny-node` to obtain data from VSS?

As I understand it, any data is obtained locally, not directly from vss. vss is simply a synchronization mechanism.

**Comment by duanyytop** [comment]
What should I do if I want to get the same activities from VSS in multi-devices?

**Comment by duanyytop** [comment]
Discussion conclusion: the `v1.10.10` of the `mutiny-wasm` doesn't include this PR and I will release `v1.10.11` to verify again.

**Comment by duanyytop** [comment]
After testing, this feature works on v1.10.11

**Pull request merged**